### ISSUE TITLE
Is the sequence of events here realistic?

### ISSUE THREAD

**Issue by domenic** [uskay/vt-demo-mpa#3]
In the current code, the sequence is:

- Intercept clicks
- Start the prerender
- Do a view transition on the source page (I think this might be a SPA view transition?)
- Wait 1.5-2 seconds
- Use `location.href` to navigate

I think in more realistic apps, the sequence is probably going to be something like:

- Start prerender earlier (e.g. mousedown, #1, or hover, #2)
- The user clicks an `<a>` (not a `<div>`) which causes the browser to start navigating
- The MPA view transition happens. (Probably with no JavaScript involved??) Since we prerendered early, the transition will be smooth, instead of having a pause in the middle while we wait for content from the network.

That's the approach I took in https://cool-massive-appendix.glitch.me/mpa-vt/ . I'd love to see a version of that which used your beautiful demo UI and transitions!

**Comment by noamr** [comment]
Yes, To @domenic's request I looked at this demo as well. I don't entirely understand the purpose of starting a same-document transition (`startViewTransition`). I would expect using regular links and perhaps prerendering on hover, without `startViewTransition`.
